### PR TITLE
[NTRN-127] chore: small improvements

### PR DESCRIPTION
**Pull request by sotnikov-s**
**task:** https://hadronlabs.atlassian.net/browse/NTRN-127

**this PR:**
- adds missing `set_contract_version` for the reserve and distribution contracts;
- optimises migration params check in `execute_migrate_from_xyk_to_cl`.

D.O.D. 6 from the task has been skipped due to majority is opposed to it.

### CODE REVIEW

**Comment by sotnikov-s** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/6589360521

**Comment by pr0n00gler** [comment]
> https://github.com/neutron-org/neutron-tests/actions/runs/6589360521

You use `main` of the integration-tests repo in your test run, but you should use `feat/xyk-to-cl-migration` since you the PR is related to the migration of liquidity

**Comment by sotnikov-s** [comment]
test run against `feat/xyk-to-cl-migration` branch:
https://github.com/neutron-org/neutron-tests/actions/runs/6614904903

test run against `feat/lockdrop-cl-pools-migration` branch:
https://github.com/neutron-org/neutron-tests/actions/runs/6612057482

File: contracts/tokenomics/reserve/src/contract.rs [hunk 1]
```diff
@@ -20,12 +20,16 @@ use cosmwasm_std::{
...
     MessageInfo, Response, StdResult, Uint128, WasmMsg,
 };
+use cw2::set_contract_version;
 use cw20::{BalanceResponse, Cw20ExecuteMsg, Cw20QueryMsg};
 use exec_control::pause::{
     can_pause, can_unpause, validate_duration, PauseError, PauseInfoResponse,
 };
 use neutron_sdk::bindings::query::NeutronQuery;
 
+pub(crate) const CONTRACT_NAME: &str = "crates.io:reserve";
```
Comment (pr0n00gler) [inline contracts/tokenomics/reserve/src/contract.rs:30]: In `Cargo.toml` we have name of the contract equals to `neutron-reserve`. Let's keep things consistent and have the same name here.

**Review by pr0n00gler — CHANGES_REQUESTED** [review]
It's out of scope, but since you've updated the reserve contract, please add similar `CONTRACT_NAME` and `CONTRACT_VERSION` constants in the distribution contract (and `cw2` methods for setting contract info)

**Comment by sotnikov-s** [comment]
test run against `feat/xyk-to-cl-migration` branch:
https://github.com/neutron-org/neutron-tests/actions/runs/6637554276

test run against `feat/lockdrop-cl-pools-migration` branch:
https://github.com/neutron-org/neutron-tests/actions/runs/6637560757

File: contracts/tokenomics/distribution/src/contract.rs [hunk 2]
```diff
@@ -22,6 +26,8 @@ pub fn instantiate(
     _info: MessageInfo,
     msg: InstantiateMsg,
 ) -> StdResult<Response> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
```
Comment (pr0n00gler) [inline contracts/tokenomics/distribution/src/contract.rs:29]: Same method must be called in migrate handler

**Review by pr0n00gler — CHANGES_REQUESTED** [review]

**Comment by sotnikov-s** [comment]
reran the above jobs:

test run against `feat/xyk-to-cl-migration` branch:
https://github.com/neutron-org/neutron-tests/actions/runs/6637554276

test run against `feat/lockdrop-cl-pools-migration` branch:
https://github.com/neutron-org/neutron-tests/actions/runs/6637560757

**Review by NeverHappened — APPROVED** [review]
Looks nice!

**Review by swelf19 — APPROVED** [review]

**Pull request merged**